packages/storage-documents/src/storage-document.ts

```diff
@@ -1,6 +1,10 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+import { ItemType } from './item-type';
+
+export type DocumentDataOnly<T> = Omit<T, keyof StorageDocument> & Partial<StorageDocument> & Partial<T>;
```

Comment:
`export type DocumentDataOnly<T extends StorageDocument> = Partial<Omit<T, keyof StorageDocument>>;`

Reply:
The intention was to keep all the required properties of the derived class required while making the properties of StorageDocument optional, and I don't think making the overall type a partial would do that

Reply:
Then it will look like this

`export type DocumentDataOnly<T extends StorageDocument> = Omit<T, keyof StorageDocument> & Partial<StorageDocument>;`

`Partial<T>` is not required if we want to have all but `StorageDocument` properties required.